bin/11_7/src/Main.cpp

```diff
@@ -0,0 +1,119 @@
+#include <algorithm>
+#include <cctype>
+#include <exception>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <utility>
+#include <vector>
+
+bool IsPunct(const unsigned char c) {
+  return c == '.' || c == ';' || c == ',' ||
+    c == '?' || c == '\'';
+}
+
+int main() {
+  try {
+    std::ifstream in { "files/In.txt" };
+    if (!in.is_open()) {
+      throw std::invalid_argument {
+        "The file \"files/File.txt\" was not found."
+      };
+    }
+
+    in >> std::noskipws;
+    std::string buffer;
+    std::copy(
+      std::istream_iterator<char> (in),
+      std::istream_iterator<char> (),
+      std::back_inserter(buffer));
+
+    // Convert all characters to lowercase.
+    std::transform(
+      std::begin(buffer),
+      std::end(buffer),
+      std::begin(buffer),
+      [] (unsigned char c) -> unsigned char {
+        return std::tolower(c);
+      });
+
+    // Replace abbreviations with the full form.
+    std::vector<std::pair<std::string, std::string>> forms {
+      { "don't", "do not" },
+      { "can't", "cannot" },
+      { "couldn't", "could not" },
+      { "mustn't", "must not" },
+      { "needn't", "need not" },
+      { "won't", "will not" },
+      { "wouldn't", "would not" },
+      { "shan't", "shall not" },
+      { "shouldn't", "should not" },
+      { "oughtn't", "ought not" },
+      { "hadn't", "had not" },
+    };
+    for (const auto& [contracted_form, long_form] : forms) {
+      while (buffer.find(contracted_form) != std::string::npos) {
```

Comment:
You need to save the last position where a short form was found, then start finding from this position.